src/event.rs

```diff
@@ -0,0 +1,530 @@
+use crate::{
+	hex_utils, ChannelManager, Error, FilesystemPersister, LdkLiteChainAccess, LdkLiteConfig,
+	NetworkGraph, PaymentInfo, PaymentInfoStorage, PaymentStatus,
+};
+
+#[allow(unused_imports)]
+use crate::logger::{
+	log_error, log_given_level, log_info, log_internal, log_trace, log_warn, FilesystemLogger,
+	Logger,
+};
+
+use lightning::chain::chaininterface::{BroadcasterInterface, ConfirmationTarget, FeeEstimator};
+use lightning::chain::keysinterface::KeysManager;
+use lightning::ln::PaymentHash;
+use lightning::routing::gossip::NodeId;
+use lightning::util::events as ldk_events;
+use lightning::util::persist::KVStorePersister;
+use lightning::util::ser::{MaybeReadable, Readable, ReadableArgs, Writeable, Writer};
+
+use bitcoin::secp256k1::Secp256k1;
+use rand::{thread_rng, Rng};
+use std::collections::{hash_map, VecDeque};
+use std::sync::{Arc, Condvar, Mutex};
+use std::thread;
+use std::time::Duration;
+
+/// The event queue will be persisted under this key.
+pub(crate) const EVENTS_PERSISTENCE_KEY: &str = "events";
+
+/// An event emitted by [`LdkLite`] that should be handled by the user. This will more often than
+/// not wrap a LDK event accessible via the `inner` field.
+#[derive(Debug, Clone)]
+pub enum LdkLiteEvent {
+	/// A payment we sent was successful.
+	PaymentSuccessful {
+		/// The hash of the payment.
+		payment_hash: PaymentHash,
```

Comment:
Well the idea of having a separate `LdkLiteEvent` struct is to provide a more user-friendly version of events that does not have to cover all LDK events, but might include other events, e.g., originating from BDK. The payment hash is included here because the top-level event should include the basic information necessary to track the payment, without the need to access `inner`. However, `inner` is provided to users that for some reason or another need to access data only provided by the wrapped event, if present.

Reply:
Hm, you mean having multiple `struct`s that all implement an `Event` trait? I don't see any real benefit from that over an `enum`. Or am I misunderstanding what you're saying?

Reply:
I honestly don't quite get the difference? `Event::FundingGenerationReady` is already a named enum struct, basically the short form of what you're suggesting? And the inner fields are also named and accessible? The only benefit I see is that it would allow to implement serialization on a per-event basis and we could store the type in a private field. Not sure that that warrants the verbosity though?

Reply:
Alright. Still not entirely sure about the benefits, but now went this way with https://github.com/lightningdevkit/ldk-lite/pull/10/commits/83124284e0c8003446982f580c7bf47c57ae43e8.

Reply:
LOL, sorry for the miscommunication, somehow didn't occur to me you could be speaking about upstream events. Just re-read the convo and it totally makes sense. I'll leave the events here wrapped for now, and will open an upstream PR to the same effect.

Reply:
See https://github.com/lightningdevkit/rust-lightning/pull/1749.